Sources/SpeziHealthKitBulkExport/SpeziHealthKitBulkExport.docc/BulkHealthExporter.md

```diff
@@ -0,0 +1,97 @@
+- A set of to-be-exported sample types.
+- A ``BatchProcessor``, which allows the app to process the individual batches of fetched samples.
+
+This structure allows the Bulk Export API to be used in a variety of ways, for different kinds of export operations. (See below for an example.)
+
+In order to perform a Health data export, an app simply calls the ``BulkHealthExporter/session(_:for:startDate:endDate:batchSize:using:startAutomatically:)`` function once at some point after the app's launch; this will either:
+- create and start a new session, if no matching session (based on the identifier) exists, or
+- restore and continue an existing session (e.g., from a previous launch of the app).
+
+It is safe to call this function multiple times and with the same input, even if a previously-created upload has already been completed.
+The session will internally keep track of its creation date, and will only ever export samples up to that date.
+This allows an app to e.g. use the `CollectSample` API to continuously fetch and collect new Health samples, and use the ``BulkHealthExporter`` to do a one-time export operation of historical Health data.
+
+The ``BulkHealthExporter/session(_:for:startDate:endDate:batchSize:using:startAutomatically:)`` function will, when a ``BulkExportSession`` is first created, also return an `AsyncStream` which can be used to access the individual results of the session's ``BatchProcessor``.
+
+- Important: Ensure that your app has sufficient HealthKit access permissions before starting bulk export sessions. The session itself will *not* prompt the user for access; instead, it will fail to fetch and process any sample types for which no HealthKit permission is granted.
+
+It is possible to ``BulkExportSession/pause()`` an export session, which can then be resumed using the ``BulkExportSession/start(retryFailedBatches:)`` function.
+
+
+### Example: Bulk-Upload of Historical Health Data to Firebase
+
+This example implements a custom ``BatchProcessor``, which uploads the exported HealthKit samples received from the ``BulkHealthExporter`` into Firebase. 
+In this case, we implicitly define the Batch Processor's `Output` type as `Void`, since we're just interested in the uploading, and don't want to perform any additional on-device operations using the results of the individual batches. 
+
+```swift
+struct FirebaseUploader: BulkHealthExporter.BatchProcessor {
+    func process<Sample>(_ samples: consuming [Sample], of sampleType: SampleType<Sample>) async throws {
+        let batch = Firestore.firestore().batch()
+        for sample in samples {
+            let document = db.collection("healthData").document(sample.uuid.uuidString) 
+            try batch.setData(from: sample.resource(), for: document)
+        }
+        try await batch.commit()
+    }
+}
+```
+
+We can then use the batch processor when creating a Bulk Export Session:
+```swift
+_ = try await bulkExporter.session(
+    "my-bulk-export-session",
```

Comment:
This example still uses a string identifier; should we demonstrate that we can have a static property as an extension to the identifier? Would be a good way to communicate the best practices.

Reply:
the docs haven't been updated yet bc i wanted to wait for your feedback on the API changes